### PR TITLE
Notes on 3rd party dependencies

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: add docs on how we manage 3rd party dependencies in the project

**Release note**:

```
dependencies documentation
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1272)

### CODE REVIEW

**Comment by jackfrancis** [comment]
@anhowe @seanknox @jchauncey FYI, a 1st draft on the language that describes how we intend to express a structured dependency management strategy

**Comment by seanknox** [comment]
This is great. Worth mentioning that all dependencies need to be registered with Microsoft OSS to be cleared for use?

File: docs/developers.md [hunk 1]
```diff
@@ -54,6 +54,14 @@ workflow for doing this is as follows:
 5. When you are ready for us to review, push your branch to GitHub, and
    then open a new pull request with us.
 
+### Third Party Dependencies
```
Comment (jchauncey) [inline docs/developers.md]: Ive added an issue about updating labels (#1277) so when people submit prs that need dependency updates/additions we can track them.

**Comment by jchauncey** [comment]
My only comment to this is I wonder if this should be some where more visible?

**Comment by jackfrancis** [comment]
@seanknox I'm inclined to defer on how much info we publish about that. Who would be the best resource to determine how we want to say that?

**Comment by seanknox** [comment]
> @seanknox I'm inclined to defer on how much info we publish about that. Who would be the best resource to determine how we want to say that?

Fair enough. As long as there's a process we (ACS) follow to ensure dependencies are registered, that's what matters.

**Pull request merged**